### PR TITLE
Debounce filtering of the store in StoreFilterField (#465)

### PR DESCRIPTION
**Pull request by haynesjm42**
#465

### CODE REVIEW

File: desktop/cmp/store/StoreFilterField.js [hunk 1]
```diff
@@ -25,11 +25,20 @@ export class StoreFilterField extends Component {
         /** Store to filter */
         store: PT.instanceOf(BaseStore).isRequired,
         /** Names of fields in store's records to filter by */
-        fields: PT.arrayOf(PT.string).isRequired
+        fields: PT.arrayOf(PT.string).isRequired,
+        /** Delay (in ms) used to buffer filtering of the store after the value changes from user input (default is 200ms). Set to 0 to filter immediately after user input. */
```
Comment (lbwexler) [inline desktop/cmp/store/StoreFilterField.js]: This long line should be broken up

File: desktop/cmp/store/StoreFilterField.js [hunk 2]
```diff
@@ -25,11 +25,20 @@ export class StoreFilterField extends Component {
...
+        filterBuffer: PT.number
     };
 
     @setter @observable value = '';
 
+    constructor(props) {
+        super(props);
+
+        const {filterBuffer = 200} = this.props;
```
Comment (lbwexler) [inline desktop/cmp/store/StoreFilterField.js]: awesome syntax here.  Seems like this should be used everywhere we currently do 
>>filterBuffer = this.props.filterBuffer || 200;
It cleanly avoids any issues of 'falsy' values not getting set....

File: desktop/cmp/store/StoreFilterField.js [hunk 3]
```diff
@@ -25,11 +25,20 @@ export class StoreFilterField extends Component {
...
+        this._debouncedFilter = debounce(this.runFilter, filterBuffer);
+    }
```
Comment (lbwexler) [inline desktop/cmp/store/StoreFilterField.js]: This kind of brings up the question I have long had about how to distiguish beween "fixed" props and "dynamic" props.  In this case,  this becomes a "fixed" prop.  Changing it after the first render of this component will have no effect.

Comment (lbwexler) [reply desktop/cmp/store/StoreFilterField.js]: (Not implying that there is a problem with that)

File: desktop/cmp/store/StoreFilterField.js (hunk 2 above)
Comment (haynesjm42) [reply desktop/cmp/store/StoreFilterField.js]: Yes definitely, been trying to make sure I use it (and lodash `isNil()`) whenever appropriate after I introduced a couple bugs in Apollo from implicit boolean conversions to falsy values..

**Pull request merged**